### PR TITLE
Redundant objc attribute

### PR DESCRIPTION
**Pull request by dirtydanee**
Implements https://github.com/realm/SwiftLint/issues/2193

### CODE REVIEW

File: Source/SwiftLintFramework/Rules/RedundantObjcAttributeRule.swift [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+public struct RedundantObjcAttributeRule: ASTRule, ConfigurationProviderRule {
+
+    public var configuration = SeverityConfiguration(.warning)
+
+    public init() {}
+
+    public static let description = RuleDescription(
+        identifier: "redundant_objc_attribute",
+        name: "Redundant @objc Attribute",
+        description: "Objective C attribute is redundant in declaration.",
```
Comment (marcelofabri) [inline Source/SwiftLintFramework/Rules/RedundantObjcAttributeRule.swift]: what do you think about: `Objective-C  attribute (@objc) is redundant in declaration.`?

File: CHANGELOG.md [hunk 2]
```diff
@@ -8,6 +8,10 @@
 
 #### Enhancements
 
+* Add `redundant_objc_attribute` to warn against declaring Objective C visibility with attributes already impling the objc attribute.
```
Comment (marcelofabri) [inline CHANGELOG.md]: this need to be wrapped in 80 chars

File: CHANGELOG.md [hunk 3]
```diff
@@ -10,6 +10,9 @@
 
 * Updates the `untyped_error_in_catch` rule to support autocorrection.  
   [Daniel Metzing](https://github.com/dirtydanee)
+* Add `redundant_objc_attribute` to warn against declaring Objective C visibility with attributes already impling the objc attribute.
```
Comment (Coder-256) [inline CHANGELOG.md]: implying.

**Comment by Coder-256** [comment]
What about `@objcMembers` classes? You should make sure that none of their members have `@objc` because it's already implied, like `@IBInspectable`, etc. except on the class level.

**Comment by dirtydanee** [comment]
@Coder-256 the `@objcMembers` are covered in the rule, like in this triggering example:
```
@objcMembers
class Foo {
   @objc ↓var bar: Any?
   @objc ↓var foo: Any?
   @objc
   class Bar {
       @objc
       var foo: Any?
   }
}
```

**Comment by dirtydanee** [comment]
@marcelofabri Are there more updates required in my implementation?

**Comment by dirtydanee** [comment]
I have managed to completely ruin this branch with my last rebase. Declining PR.

**Pull request closed without merging**